### PR TITLE
[IOPID-602] : Adjustment for build static export

### PR DESCRIPTION
**Pull request by claudiodall**
## Short description

The need arose to modify the project so that it could correctly generate the export of the project into static files (with internalization). In the current mode, it was being compiled as a Static Site Generator (SSG), project exported with (html, css, js files).

Note: With this approach, there will be no way to identify the default language, but there will be the path prefix /it/ or /lang/.

In comparison to what is described in the next-intl documentation

https://next-intl-docs.vercel.app/docs/routing/middleware#usage-without-middleware-static-export

This way, the export was resolved to obtain the desired static package with HTML and JavaScript files.
## List of changes proposed in this pull request

- Stored landing locale in local storage (necessary for redirect callback after spid login)
- Adjustment according to next-intl documentation for static next export
- Layout file modification according to next export docs
- Usage of next-intl components that wrap other components like LINK, useRouter to use relative paths with the locale
- Appropriate next-config
- Removal of middleware not supported in SSG export mode
- Other adjustments

## How to test

Generate the project in static mode using yarn build. Once the project is created in the .out folder, verify the presence of HTML files. Launch the build using yarn run start-static and verify the correct functioning of the portal.

### CODE REVIEW

File: src/app/[locale]/(pages)/_templateServer/page.tsx
Comment (Ladirico) [inline src/app/[locale]/(pages)/_templateServer/page.tsx:1]: I think you can delete this page

File: src/app/[locale]/_component/profileCards/profileCards.tsx [hunk 1]
```diff
@@ -49,15 +50,11 @@ const ProfileCards = (): React.ReactElement => {
...
-
-              <ButtonNaked
-                href={ROUTES.PROFILE_BLOCK}
-                color="primary"
-                endIcon={<ArrowForwardIcon />}
-                size="medium"
-              >
-                {t('profile.lockaccess')}
-              </ButtonNaked>
+              <Link href={ROUTES.PROFILE_BLOCK}>
```
Comment (Ladirico) [inline src/app/[locale]/_component/profileCards/profileCards.tsx:53]: Here and in the other occurrences we can eliminate this link that wraps the button and directly insert the onclick on the button with the router.push (so, using the userRouter)

**Review by Ladirico — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -11,31 +11,36 @@ This repository contains the code that composes the io-web-profile front-end.
...
-ENDPOINT_ERROR=http://localhost:3000/accedi/errore
-ENDPOINT_SUCCESS=http://localhost:3000/
+To ensure proper execution of the service, please follow the documentation provided by hub-spid-login. Additionally, make the following modifications to the default values within the hub-spid-login .env file:
 
-Note: The port used here is the same one used to expose the web app.
+``ENDPOINT_ERROR=http://localhost:3000/accedi/errore``
+``ENDPOINT_SUCCESS=http://localhost:3000/it/accedi
+``
+``
+ENABLE_JWT=true``
```
Comment (Ladirico) [inline README.md:34]: you can add a major level of description, adding some like this: in order to enable JWTs we recommend following this part of the documentation https://github.com/pagopa/hub-spid-login-ms#jwt-support

File: src/app/[locale]/(pages)/blocco-accesso/page.tsx [hunk 3]
```diff
@@ -32,9 +33,11 @@ const ProfileBlock = (): React.ReactElement => {
         <Grid sx={{ maxWidth: '576px' }}>
           {isIDPKnown && <IdpListOnApp />}
           <Typography mb={5}>{t('common.lockaccessinfo')}</Typography>
-          <Button href={ROUTES.PROFILE_BLOCK_SUCCESS} variant="contained" size="medium">
-            {t('profile.lockaccess')}
-          </Button>
+          <Link href={ROUTES.PROFILE_BLOCK_SUCCESS}>
```
Comment (Ladirico) [inline src/app/[locale]/(pages)/blocco-accesso/page.tsx:36]: Here and in the other occurrences we can eliminate this link that wraps the button and directly insert the onclick on the button with the router.push (so, using the userRouter)

**Review by Ladirico — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(pages)/_templateServer/page.tsx
Comment (claudiodall) [reply src/app/[locale]/(pages)/_templateServer/page.tsx:1]: Sure

File: README.md (hunk 2 above)
Comment (claudiodall) [reply README.md:34]: Descrption changed

File: src/app/[locale]/(pages)/blocco-accesso/page.tsx (hunk 3 above)
Comment (claudiodall) [reply src/app/[locale]/(pages)/blocco-accesso/page.tsx:36]: Done

**Pull request merged**